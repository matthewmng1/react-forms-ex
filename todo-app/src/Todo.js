import React from "react";

const Todo = ({id, todo, handleRemove}) => {
  const removeTodo = () => handleRemove(id);

  return (
    <div>
      <li>{todo} <button onClick={removeTodo}>x</button> 
</li>
    </div>
  );
}

export default Todo;
