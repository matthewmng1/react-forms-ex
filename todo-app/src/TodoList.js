import React, {useState} from "react"
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
  const [todos, setTodos] = useState([])

  const addTodo = (newTodo) => {
    setTodos(todos => [...todos, newTodo])
  }

  const removeTodo = (id) => {
    setTodos(todos => todos.filter(todo => todo.id !== id))
  }

  const tasks = todos.map(todo => (
    <Todo 
    key = {todo.id}
    id = {todo.id}
    todo = {todo.todo}
    handleRemove={removeTodo}/>
  ))

  return (
    <div>
      <h1>To-do List</h1>
      <NewTodoForm createTodo={addTodo}/>
      {tasks}
    </div>
  )
}

export default TodoList;