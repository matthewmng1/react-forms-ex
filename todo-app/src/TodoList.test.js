import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";
import Todo from "./Todo";

it(`renders without crashing`, () => {
  render (<TodoList />);
});

it(`matches snapshop`, () => {
  const {asFragment} = render(<TodoList /> );
  expect(asFragment()).toMatchSnapshot();
})

it(`adds a new task to the todo list`, () => {
  const { queryByText, getByLabelText} = render (<TodoList />);
  const input = getByLabelText("To-do List")
  const btn = queryByText("Add Todo")
  expect(queryByText("Get milk")).not.toBeInTheDocument();;
  fireEvent.change(input, {target: {value: 'Get milk'}});
  fireEvent.click(btn);
  expect(queryByText("Get milk")).toBeInTheDocument();;
})