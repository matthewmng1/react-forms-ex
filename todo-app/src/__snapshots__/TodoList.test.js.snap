// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshop 1`] = `
<DocumentFragment>
  <div>
    <h1>
      To-do List
    </h1>
    <div>
      <form>
        <label
          for="todo"
        >
          To-do List
        </label>
        <br />
        <input
          id="todo"
          name="todo"
          placeholder="New Task"
          type="text"
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
